### ISSUE TITLE
Sort dataframe using Date32 column is causing nulls to appear

### ISSUE THREAD

**Issue by AdrianAntico** [pola-rs/polars#1277]
#### Are you using Python or Rust?
Python

#### What version of polars are you using?
0.8.27-beta.1

#### What operating system are you using polars on?
Edition	Windows 10 Pro
Version	20H2
Installed on	‎3/‎12/‎2021
OS build	19042.1165
Experience	Windows Feature Experience Pack 120.2212.3530.0

#### Describe your bug.
When I try to sort a DataFrame (panel style data) using the Date32 column to sort by, causes null values to appear in the Date column used to sort.

#### What are the steps to reproduce the behavior?

```
import polars as pl
from polars import 
data = pl.read_csv('https://www.dropbox.com/s/2str3ek4f4cheqi/walmart_train.csv?dl=1')
data['Date'] = data['Date'].cast(pl.Date32)
print(data)
>>> print(data)
shape: (421570, 5)
┌───────┬──────┬──────────────┬──────────────┬───────────┐
│ Store ┆ Dept ┆ Date         ┆ Weekly_Sales ┆ IsHoliday │
│ ---   ┆ ---  ┆ ---          ┆ ---          ┆ ---       │
│ i64   ┆ i64  ┆ date32(days) ┆ f64          ┆ bool      │
╞═══════╪══════╪══════════════╪══════════════╪═══════════╡
│ 1     ┆ 1    ┆ 2010-02-05   ┆ 2.49245e4    ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 1    ┆ 2010-02-12   ┆ 4.603949e4   ┆ true      │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 1    ┆ 2010-02-19   ┆ 4.159555e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 1    ┆ 2010-02-26   ┆ 1.940354e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ ...   ┆ ...  ┆ ...          ┆ ...          ┆ ...       │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 45    ┆ 98   ┆ 2012-09-28   ┆ 508.37       ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 45    ┆ 98   ┆ 2012-10-05   ┆ 628.1        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 45    ┆ 98   ┆ 2012-10-12   ┆ 1061.02      ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 45    ┆ 98   ┆ 2012-10-19   ┆ 760.01       ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 45    ┆ 98   ┆ 2012-10-26   ┆ 1076.8       ┆ false     │
└───────┴──────┴──────────────┴──────────────┴───────────┘

x = data.clone()
x.sort('Date', reverse = True, in_place = True)
>>> x
shape: (421570, 5)
┌───────┬──────┬──────────────┬──────────────┬───────────┐
│ Store ┆ Dept ┆ Date         ┆ Weekly_Sales ┆ IsHoliday │
│ ---   ┆ ---  ┆ ---          ┆ ---          ┆ ---       │
│ i64   ┆ i64  ┆ date32(days) ┆ f64          ┆ bool      │
╞═══════╪══════╪══════════════╪══���═══════════╪═══════════╡
│ 1     ┆ 1    ┆ 2012-10-26   ┆ 2.739081e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 2    ┆ 2012-10-26   ┆ 4.313488e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 3    ┆ 2012-10-26   ┆ 9350.9       ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 4    ┆ 2012-10-26   ┆ 3.62926e4    ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ ...   ┆ ...  ┆ ...          ┆ ...          ┆ ...       │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 18.05        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 37.83        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 53.96        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 77.95        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 32.73        ┆ false     │
└───────┴──────┴──────────────┴──────────────┴───────────┘

x = data.clone()
x = x.sort('Date', reverse = True)
>>> x
shape: (421570, 5)
┌───────┬──────┬──────────────┬──────────────┬───────────┐
│ Store ┆ Dept ┆ Date         ┆ Weekly_Sales ┆ IsHoliday │
│ ---   ┆ ---  ┆ ---          ┆ ---          ┆ ---       │
│ i64   ┆ i64  ┆ date32(days) ┆ f64          ┆ bool      │
╞═══════╪══════╪══════════════╪══���═══════════╪═══════════╡
│ 1     ┆ 1    ┆ 2012-10-26   ┆ 2.739081e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 2    ┆ 2012-10-26   ┆ 4.313488e4   ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 3    ┆ 2012-10-26   ┆ 9350.9       ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 1     ┆ 4    ┆ 2012-10-26   ┆ 3.62926e4    ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ ...   ┆ ...  ┆ ...          ┆ ...          ┆ ...       │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 18.05        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 37.83        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 53.96        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 77.95        ┆ false     │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┤
│ 44    ┆ 52   ┆ null         ┆ 32.73        ┆ false     │
└───────┴──────┴──────────────┴──────────────┴───────────┘
```

**Comment by ritchie46** [comment]
Hi @AdrianAntico, this problem seems to be resolved in version `0.9.1`. Could you update and see if your problem persists?

**Comment by AdrianAntico** [comment]
@ritchie46 Thanks, the update did the trick!
